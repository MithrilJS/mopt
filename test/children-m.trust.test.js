"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("Nested m.trust()", () => {
        it("should support nested m.trust() invocations", () =>
            expect(
                code(`m("1", m.trust("1"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should support a single-item array of nested m.trust() invocations", () =>
            expect(
                code(`m("1", [ m.trust("2") ])`)
            )
            .toMatchSnapshot()
        );

        it("should support a multi-item array of nested m.trust() invocations", () =>
            expect(
                code(`m("1", [ m.trust("2"), m.trust("3") ])`)
            )
            .toMatchSnapshot()
        );

        it("should support arrays of nested m.trust() invocations & plain nodes", () =>
            expect(
                code(`m("1", [ m("2"), "3" ])`)
            )
            .toMatchSnapshot()
        );
        
        it("should support multiple nested m.trust() invocations", () =>
            expect(
                code(`m("1", m.trust("2"), m.trust("3"), m.trust("4"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should support attrs + nested m.trust() invocations", () =>
            expect(
                code(`m("1", { title : "bar" }, m.trust("2"))`)
            )
            .toMatchSnapshot()
        );
        
        it("should supported attrs + arrays of nested m.trust() invocations", () =>
            expect(
                code(`m("1", { title : "bar" }, [ m.trust("2") ])`)
            )
            .toMatchSnapshot()
        );
        
        it("should support attrs + multiple nested m.trust() invocations", () =>
            expect(
                code(`m("1", { title : "bar" }, m.trust("2"), m.trust("3"), m.trust("4"))`)
            )
            .toMatchSnapshot()
        );
    });
});
