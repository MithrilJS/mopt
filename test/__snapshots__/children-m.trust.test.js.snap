// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children Nested m.trust() should support a multi-item array of nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,undefined,[m.vnode(\\"<\\",undefined,undefined,\\"2\\",undefined,undefined),m.vnode(\\"<\\",undefined,undefined,\\"3\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should support a single-item array of nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,undefined,[m.vnode(\\"<\\",undefined,undefined,\\"2\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should support arrays of nested m.trust() invocations & plain nodes 1`] = `"m.vnode(\\"1\\",undefined,undefined,[m.vnode(\\"2\\",undefined,undefined,[],undefined,undefined),m.vnode(\\"#\\",undefined,undefined,\\"3\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should support attrs + multiple nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,{title:\\"bar\\"},[m.vnode(\\"<\\",undefined,undefined,\\"2\\",undefined,undefined),m.vnode(\\"<\\",undefined,undefined,\\"3\\",undefined,undefined),m.vnode(\\"<\\",undefined,undefined,\\"4\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should support attrs + nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,{title:\\"bar\\"},[m.vnode(\\"<\\",undefined,undefined,\\"2\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should support multiple nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,undefined,[m.vnode(\\"<\\",undefined,undefined,\\"2\\",undefined,undefined),m.vnode(\\"<\\",undefined,undefined,\\"3\\",undefined,undefined),m.vnode(\\"<\\",undefined,undefined,\\"4\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should support nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,undefined,[m.vnode(\\"<\\",undefined,undefined,\\"1\\",undefined,undefined)],undefined,undefined);"`;

exports[`Children Nested m.trust() should supported attrs + arrays of nested m.trust() invocations 1`] = `"m.vnode(\\"1\\",undefined,{title:\\"bar\\"},[m.vnode(\\"<\\",undefined,undefined,\\"2\\",undefined,undefined)],undefined,undefined);"`;
